EndlessClient/Dialogs/GuildDialog.cs

```diff
@@ -1,233 +1,522 @@
+            );
+        }
+
+        private void SetupWaitingForMembersState()
+        {
+            AddItemToList(new ListDialogItem(this, ListDialogItem.ListItemStyle.Small, 0)
+            {
+                PrimaryText = _localizedStringFinder.GetString(EOResourceID.GUILD_PLEASE_WAIT_FOR_ALL_MEMBERS_TO_JOIN)
+            }, false);
+
+            AddItemToList(new ListDialogItem(this, ListDialogItem.ListItemStyle.Small, 1), false);
+
+            AddItemToList(new ListDialogItem(this, ListDialogItem.ListItemStyle.Small, 0)
+            {
+                PrimaryText = Capitalize(_characterProvider.MainCharacter.Name)
+            }, false);
+        }
+
+        private void SetStateIfInGuild(State state)
         {
             if (!_characterProvider.MainCharacter.InGuild)
             {
-                var dlg = _messageBoxFactory.CreateMessageBox(DialogResourceID.GUILD_NOT_IN_GUILD);
+                var dlg = _messageBoxFactory.CreateMessageBox(DialogResourceID.GUILD_ALREADY_A_MEMBER);
                 dlg.ShowDialog();
                 return;
             }
 
             SetState(state);
         }
+
+        private void SetStateIfNotInGuild(State state)
+        {
+            if (_characterProvider.MainCharacter.InGuild)
+            {
+                var dlg = _messageBoxFactory.CreateMessageBox(DialogResourceID.GUILD_ALREADY_A_MEMBER);
+                dlg.ShowDialog();
+                return;
+            }
+
+            SetState(state);
+        }
+
+        private void ShowJoinGuildMessageBox()
+        {
+            if (_characterProvider.MainCharacter.InGuild)
+            {
+                var dlgAlreadyMember = _messageBoxFactory.CreateMessageBox(DialogResourceID.GUILD_ALREADY_A_MEMBER);
+                dlgAlreadyMember.ShowDialog();
+                return;
+            }
+
+            var dlgJoin = _textMultiInputDialogFactory.Create(
+                _localizedStringFinder.GetString(DialogResourceID.GUILD_JOIN_GUILD),
+                _localizedStringFinder.GetString(DialogResourceID.GUILD_JOIN_GUILD + 1),
+                TextMultiInputDialog.DialogSize.Two,
+                new TextMultiInputDialog.InputInfo { Label = _localizedStringFinder.GetString(EOResourceID.GUILD_GUILD_TAG), MaxChars = 3 },
+                new TextMultiInputDialog.InputInfo { Label = _localizedStringFinder.GetString(EOResourceID.GUILD_RECRUITER), MaxChars = 12 }
+            );
+
+            dlgJoin.DialogClosing += (sender, e) =>
+            {
+                if (e.Result == XNADialogResult.OK)
+                {
+                    var guildTag = dlgJoin.Responses[0];
+                    var recruiterName = dlgJoin.Responses[1];
+
+                    if (guildTag.Length == 0)
+                    {
+                        e.Cancel = true;
+                        var dlg = _messageBoxFactory.CreateMessageBox(DialogResourceID.GUILD_CREATE_TAG_FIELD_EMPTY);
+                        dlg.Show();
```

Comment:
I think you want `ShowDialog` on these? I think `Show` is the awaitable method signature and should almost never be used